src/index.ts

```diff
@@ -529,7 +529,7 @@ export async function findDefinition(device: Zh.Device, generateForUnknown = fal
         }
 
         // Do not add this definition to cache, as device configuration might change.
-        const {definition} = await generateDefinition(device);
+        const {definition} = device.type === "GreenPower" ? generateGreenPowerDefinition(device) : await generateDefinition(device);
```

Comment:
I think it is better to push this if statement into `generateDefinition` (no need to do it twice in this file)